src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -267,6 +269,33 @@ public void moveToDefaultDirectory() {
         }
     }
 
+    /**
+     * Gets the filename for the current linked file and compares it to the new suggested filename.
+     * @return true if the suggested filename is same as current filename.
+     */
+    public boolean isGeneratedNameSameAsOriginal() {
+
+        Path file = Paths.get(this.linkedFile.getLink());
+        String currentFileName = file.getFileName().toString();
+        String suggestedFileName = this.linkedFileHandler.getSuggestedFileName();
+
+        return currentFileName.equals(suggestedFileName);
+    }
+
+    /**
+     * Compares suggested filepath of current linkedFile with existing filepath.
+     * @return true if suggested filepath is same as existing filepath.
+     */
+    public boolean isGeneratedPathSameAsOriginal() {
+
+        Optional<Path> newDir = databaseContext.getFirstExistingFileDir(filePreferences);
+
+        Optional<Path> currentDir = linkedFile.findIn(databaseContext, filePreferences);
+
+        BiPredicate<Path, Path> equality = Files::isSameFile;
```

Comment:
I am not sure how to handle the exception here

Reply:
Could you be a bit  more precise what and where an exception is thrown?

Reply:
`Files::isSameFile` unhandled IOException is shown. I tried including in the try catch but the error still persists

Reply:
It should be something like
```
equality = (fileA, fileB) -> {
    throw {
       return Files.isSameFile(fileA, fileB);
   } catch (IOException ex) {
       return false;
   }
}
```